### PR TITLE
Bugfix FXIOS-5615 [v110] Fix sponsored tile favicons

### PR DESCRIPTION
**Pull request by OrlaM**
[FXIOS-5615](https://mozilla-hub.atlassian.net/browse/FXIOS-5615)
#12975

### CODE REVIEW

**Comment by lmarceau** [comment]
Even with this change I still get this sponsored tile as a default favicon. The letter change to "H" instead of "T".
Screen Shot 2023-01-18 at 3 45 41 PM

**Comment by OrlaM** [comment]
I think maybe I'm not seeing it because the ones I'm seeing are also bundled icons. I don't see Trivago, is there any way to force which sponsored tiles show?

**Comment by lmarceau** [comment]
Just for others that might be reading this, through VPN we can change location and this should trigger different sponsored tiles.

**Comment by isabelrios** [comment]
FYi: The warnings went up to 34 and so the build failed: https://mozilla.slack.com/archives/CAFC45W5A/p1674146499603069

File: BrowserKit/Sources/SiteImageView/SiteImageHandler.swift [hunk 1]
```diff
@@ -52,6 +52,10 @@ public class DefaultSiteImageHandler: SiteImageHandler {
                                                    usesIndirectDomain: usesIndirectDomain)
         }
 
+        if usesIndirectDomain, type == .favicon {
+            imageModel.faviconURL = URL(string: urlStringRequest)
+        }
```
Comment (lmarceau) [inline BrowserKit/Sources/SiteImageView/SiteImageHandler.swift:57]: Wondering if it's worth adding a comment here?

**Review by lmarceau — APPROVED** [review]

**Comment by OrlaM** [comment]
Build is green on BR 
image

**Comment by OrlaM** [comment]
@Mergifyio backport release/v110.0

**Pull request merged**